ansible/playbooks/adhoc/grow_root_vg/grow_root_vg.yml

```diff
@@ -0,0 +1,132 @@
+---
+# This playbook grows the root VG on a node by:
+#  * expanding /dev/xvda
+#  * using growpart to expand the rootvg partition
+#  * grow /var to 64GB
+#
+#  To run:
+#  1. Source your AWS credentials (make sure it's the corresponding AWS account) into your environment
+#    export AWS_PROFILE=<aws account>
+#    export AWS_DEFAULT_REGION=<aws region>
+#
+# 2. run the playbook:
+#   ansible-playbook -e 'cli_tag_name=<tag-name>' grow_root_vg.yml
+#
+#  Example:
+#   ansible-playbook -e 'cli_tag_name=ops-compute-12345' grow_root_vg.yml
+#
+#  Notes:
+#  * By default this will do a900GB GP2 volume.  The can be overidden with the "-e 'cli_volume_size=something'" variable
+#  * This can be done with NO downtime on the host, but you probably ought to test a reboot 
+#
+
+- name: Grow the rootvg volume group
+  hosts: "oo_name_{{ cli_tag_name }}"
+  user: root
+  connection: ssh
+  gather_facts: no
+
+  vars:
+    cli_volume_size: 90
+    cli_var_size: 64
+
+  pre_tasks:
+  - fail:
+      msg: "This playbook requires {{item}} to be set."
+    when: "{{ item }} is not defined or {{ item }} == ''"
+    with_items:
+    - cli_tag_name
+    - cli_volume_size
+    - cli_var_size
+
+  - name: Attempt to find the Phyisical Volume that rootvg is using
+    shell: "pvs | awk '/rootvg/ {print $1}'" 
+    register: rootvg_pv
+    ignore_errors: yes
+
+  - debug:
+      var: rootvg_pv
+
+  - name: fail if we don't find a single rootvg physical volume
+    fail:
+      msg:  Unable to find single rootvg physical volume. Please investigate manually.
+    when: rootvg_pv.stdout_lines|length != 1
+
+  - name: get list of volumes from AWS
+    delegate_to: localhost
+    ec2_vol:
+      state: list
+      instance: "{{ ec2_id }}"
+      region: "{{ ec2_region }}"
```

Comment:
you can add `profile: "{{ oo_profile }}"` here to avoid having to export ENV vars above.

Reply:
I wasn't able to find a reference to oo_profile in any of our code, curious what this does. I added {{ ec2_region }} to the aws resize command, so all you have to set is AWS_PROFILE